nca/CoreDS/ConnectivityProperties.py

```diff
@@ -0,0 +1,634 @@
+                                           src_peers_interval=src_peers_interval,
+                                           dst_peers_interval=base_peer_set.get_peer_interval_of(PeerSet({peer})),
+                                           base_peer_set=base_peer_set)
+            conn_properties |= props
+        return conn_properties
+
+    @staticmethod
+    def make_conn_props_from_dict(peer_container, cube_dict):
+        """
+        Create ConnectivityProperties from the given cube
+        :param PeerContainer peer_container: the set of all peers
+        :param dict cube_dict: the given cube represented as a dictionary
+        :return: ConnectivityProperties
+        """
+        cube_dict_copy = cube_dict.copy()
+        src_ports = cube_dict_copy.pop("src_ports", ConnectivityProperties.get_full_dim_having_orig_type("src_ports"))
+        dst_ports = cube_dict_copy.pop("dst_ports", ConnectivityProperties.get_full_dim_having_orig_type("dst_ports"))
+        protocols = cube_dict_copy.pop("protocols", ConnectivityProperties.get_full_dim_having_orig_type("protocols"))
+        src_peers = cube_dict_copy.pop("src_peers", ConnectivityProperties.get_full_dim_having_orig_type("src_peers"))
+        dst_peers = cube_dict_copy.pop("dst_peers", ConnectivityProperties.get_full_dim_having_orig_type("dst_peers"))
+        paths_dfa = cube_dict_copy.pop("paths", ConnectivityProperties.get_full_dim_having_orig_type("paths"))
+        hosts_dfa = cube_dict_copy.pop("hosts", ConnectivityProperties.get_full_dim_having_orig_type("hosts"))
+        methods = cube_dict_copy.pop("methods", ConnectivityProperties.get_full_dim_having_orig_type("methods"))
+        icmp_type_interval = cube_dict_copy.pop("icmp_type", ConnectivityProperties.get_full_dim_having_orig_type("icmp_type"))
+        icmp_code_interval = cube_dict_copy.pop("icmp_code", ConnectivityProperties.get_full_dim_having_orig_type("icmp_code"))
+        assert not cube_dict_copy
+        return ConnectivityProperties.make_conn_props(peer_container, src_ports=src_ports, dst_ports=dst_ports,
+                                                      protocols=protocols, src_peers=src_peers, dst_peers=dst_peers,
+                                                      paths_dfa=paths_dfa, hosts_dfa=hosts_dfa, methods=methods,
+                                                      icmp_type_interval=icmp_type_interval,
+                                                      icmp_code_interval=icmp_code_interval)
+
+    @staticmethod
+    def make_empty_props():
+        """
+        Returns empty connectivity properties, representing logical False
+        :return: ConnectivityProperties
+        """
+        return ConnectivityProperties(create_empty=True)
+
+    @staticmethod
+    def make_all_props():
+        """
+        Returns all connectivity properties, representing logical True
+        :return: ConnectivityProperties
+        """
+        return ConnectivityProperties()
```

Comment:
Is it valid to create `all_props` with an empty `base_peer_set`?

Reply:
Added flag `create_all `to optimize `make_all_props`.
It is valid to create it with an empty `base_peer_set`, because the `src_peers `and `dst_peers `dimensions do not contain peers.